admin/common_auth/admin.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib import admin
+from django.contrib.auth.admin import UserAdmin
+from django.contrib.auth.models import Permission
+from .models import MyUser
+
+from django.contrib.auth.forms import PasswordResetForm, UserCreationForm
+
+class PermissionAdmin(admin.ModelAdmin):
+    search_fields = ['name', 'codename']
+
+class CustomUserRegistrationForm(UserCreationForm):
+    class Meta:
+            model = MyUser
+            fields = ['password', 'first_name', 'last_name', 'email', 'is_active', 'is_staff',
+            'is_superuser', 'groups', 'user_permissions', 'last_login', ]
+    def __init__(self, *args, **kwargs):
+        super(CustomUserRegistrationForm, self).__init__(*args, **kwargs)
+        self.fields['first_name'].required = True
+        self.fields['last_name'].required = True
+
+class CustomUserAdmin(UserAdmin):
+    add_form = CustomUserRegistrationForm
+    list_display = ['email', 'first_name', 'last_name', 'is_active']
+    fieldsets = (
+        (None, {'fields': ('email', 'password',)}),
+        ('Personal info', {'fields': ('first_name', 'last_name', 'email', 'date_joined', 'last_login',)}),
+        ('Permissions', {'fields': ('is_active', 'is_staff', 'is_superuser', 'groups', 'user_permissions',)}),
+    )
+    add_fieldsets = (
+        (None, {'fields':
+                ('email', 'first_name', 'last_name', 'password1', 'password2'),
+                }),)
+    search_fields = ('email', 'first_name', 'last_name',)
+    ordering = ('last_name', 'first_name',)
+    actions = ['send_email_invitation']
+
+    def send_email_invitation(self, request, queryset):
+        for user in queryset:
+            if user.is_active is False:  # email doesn't send unless user is active
```

Comment:
Loop through all the selected users, If the user is not active make them active and save them. Create a reset form for their user, save the form, deactivate them, and save them again. By setting them to is_active = False regardless of what they used to be you're disabling everyone you send invitations to. Is that intended functionality?